### PR TITLE
Remove leading and trailing spaces before saving custom filter keyword

### LINKED ISSUE #23789: Trailing white spaces can make filters not work
### Steps to reproduce the problem

1. Create a new filter in Settings → Filters  and deliberately add a trailing white space
2. Save the filter
3. Check your Timeline and if the filter is hiding posts containing the filtered words

### Expected behaviour

All posts with the words defined in the filter should be hidden.

### Actual behaviour

Some posts will still appear. 

### Detailed description

This looks like its obvious since of course the filter expression is taken literally. The filter will sometimes work, (when the filter expression is in a middle or beginning of a sentence) and sometimes not (for example when the word you actually want to block is followed by punctuation). 

However a trailing white space is easy to overlook and I already had it happen to me several times so that I thought the filter wasn't working at all. That the culprit is a trailing (or leading) white space is not obvious to the user and the resulting behavior confusing. 

I suggest to either make white space characters visible or trimming the filter expression when saving the filter to prevent this from happening. 

### Specifications

Mastodon v4.1.0

### PR DESCRIPTION
**Pull request by iamradioactive**
Fixes #23789

### CODE REVIEW

File: app/models/custom_filter_keyword.rb [hunk 1]
```diff
@@ -23,6 +23,10 @@ class CustomFilterKeyword < ApplicationRecord
   before_destroy :prepare_cache_invalidation!
   after_commit :invalidate_cache!
 
+  def keyword=(value)
+    super(value&.strip)
```
Comment (Gargron) [inline app/models/custom_filter_keyword.rb:27]: You can use a `before_validation` callback instead of overriding the setter.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/models/custom_filter_keyword.rb (hunk 1 above)
Comment (iamradioactive) [reply app/models/custom_filter_keyword.rb:27]: @Gargron, I was thinking along the same lines untilI I came across the following thread. I also found a similar usage in the repository so tried to stick to the same conventions.

Thread: https://stackoverflow.com/a/11071577/7768631
Reference: https://github.com/mastodon/mastodon/blob/main/app/models/account_migration.rb#L55

Comment (iamradioactive) [reply app/models/custom_filter_keyword.rb:27]: @Gargron , The ideal place to validate this is indeed a `before_validation`. Let me know if you still prefer the same. I can create a separate PR to make the changes in other places as well, to keep things uniform throughout the codebase.

**Comment by roycewilliams** [comment]
In the absence of regular expressions, treating leading/trailing space as significant, when combined with disabling "whole word", can be a pretty powerful filtering option that I would hate to lose.

As a compromise, could the spaces just be made visually obvious somehow? This would let people quickly determine the root cause of the filter problem when spaces are *accidental*, while still allowing people to include leading/trailing space *on purpose* if desired.

**Comment by roycewilliams** [comment]
If you can, please highlight this change prominently as a compatibility / functionality change in the release notes. Stripping whitespace from a filter term with "whole word" unchecked can significantly broaden matches - and if "hide completely "is selected for that filter, the user might not notice for quite some time.

**Pull request closed without merging**